include/dsAudio.h

```diff
@@ -2063,6 +2063,30 @@ dsError_t  dsGetSecondaryLanguage(intptr_t handle, char* sLang);
  */
 dsError_t dsGetHDMIARCPortId(int *portId);
 
+/**
+* @brief Sets the Mixer Volume level for the MS12 input
+*
+* This function sets the MS12 volume level for either primary/main or system audio input
+*
+* @param[in] handle  - Handle for the output Audio port
+* @param[in] aInput  - dsAudioInputPrimary / dsAudioInputSystem. Please refer ::dsAudioInput_t
+* @param[in] volume  - volume to be set (0 to 100)
+*
+* @return dsError_t                        - Status
+* @retval dsERR_NONE                       - Success
+* @retval dsERR_NOT_INITIALIZED            - Module is not initialised
+* @retval dsERR_INVALID_PARAM              - Parameter passed to this function is invalid
+* @retval dsERR_OPERATION_NOT_SUPPORTED    - The attempted operation is not supported; e.g: source devices
+* @retval dsERR_OPERATION_FAILED           - The attempted operation has failed
+*
+* @pre dsAudioPortInit() and and dsGetAudioPort() must be called before calling this API
+*
+* @warning  This API is Not thread safe
+*
+*
+*/
+dsError_t dsSetAudioMixerLevels (intptr_t handle, dsAudioInput_t aInput, int volume);
```

Comment:
is this API need to support both sink &  source ?

Reply:
For now it is implemented for sink device.

Reply:
it's not about the current implementation, if any new vendor need to support this API for sink & source

Reply:
Where is the dsGetMs12AudioMixerLevels()?

Reply:
If this function is specifically for sink then the documenation should method this

Reply:
Get API is not required as AS is managing and persisting these settings.   At the moment, set API should be good enough for unblocking the feature

Reply:
This API is specific to sink

Reply:
please mention same in @brief

Reply:
Addressed